src/body_operations.cpp

```diff
@@ -116,7 +127,9 @@ Body* constructBodyFromMsgHelper(const T& shape_msg, const geometry_msgs::Pose&
         q = Eigen::Quaterniond(1.0, 0.0, 0.0, 0.0);
       }
       Eigen::Isometry3d af(Eigen::Translation3d(pose.position.x, pose.position.y, pose.position.z) * q);
-      body->setPose(af);
+      body->setPoseDirty(af);
+      body->setDimensionsDirty(shape);
```

Comment:
Same here, `setDimensions()` wasn't called before.

Reply:
As above, no semantics is changed (the original `createBodyFromShape()` would call `setDimensions()` internally). It just allows utilizing the speedup using `*Dirty` functions.